ReactWindows/ReactNative/Touch/TouchHandler.cs

```diff
@@ -176,46 +151,68 @@ private int IndexOfPointerWithId(uint pointerId)
             return -1;
         }
 
-        private UIElement GetReactViewTarget(DependencyObject originalSource, Point point)
+        private static DependencyObject GetReactViewTarget(PointerRoutedEventArgs e, UIElement view)
         {
-            // If the target is not a child of the root view, then this pointer
-            // event does not belong to React.
-            if (!RootViewHelper.IsReactSubview(originalSource))
+            // Ensure the original source is a DependencyObject
+            var originalSource = e.OriginalSource as DependencyObject;
+            if (originalSource == null)
             {
                 return null;
             }
 
-            var adjustedPoint = AdjustPointForStatusBar(point);
-            var sources = VisualTreeHelper.FindElementsInHostCoordinates(adjustedPoint, _view);
-
-            // Get the first React view that does not have pointer events set
-            // to 'none' or 'box-none', and is not a child of a view with 
-            // 'box-only' or 'none' settings for pointer events.
+            // Get the React view hierarchy from the original source.
+            var hierarchy = RootViewHelper.GetReactViewHierarchy(originalSource);
+            var enumerator = hierarchy.GetEnumerator();
 
-            // TODO: use pooled data structure
-            var isBoxOnlyCache = new Dictionary<DependencyObject, bool>();
-            foreach (var source in sources)
+            // Try to get the first React view.
+            if (!enumerator.MoveNext())
             {
-                if (!source.HasTag())
-                {
-                    continue;
-                }
+                return null;
+            }
 
-                var pointerEvents = source.GetPointerEvents();
-                if (pointerEvents == PointerEvents.None || pointerEvents == PointerEvents.BoxNone)
+            // If the first React view has `pointerEvents: box-none`, revert
+            // to using the `VisualTreeHelper` to find potentially occluded views.
+            // This condition should be rare.
+            if (enumerator.Current.GetPointerEvents() == PointerEvents.BoxNone)
+            {
+                // The pointer entered and exited events don't bubble, so they
+                // are subscribed at the granular level of views. They use this
+                // helper but pass in a null value for the root view, so we
+                // must determine what the root view is here.
+                var root = view ?? RootViewHelper.GetRootView(originalSource);
+                var rootPoint = e.GetCurrentPoint(root);
+                var transform = root.TransformToVisual(Window.Current.Content);
+                var point = transform.TransformPoint(rootPoint.Position);
+                var adjustedPoint = AdjustPointForStatusBar(point);
+
+                hierarchy = VisualTreeHelper.FindElementsInHostCoordinates(adjustedPoint, root)
```

Comment:
>hierarchy [](start = 16, length = 9)

Wrong hierarchy. Grab the first non-`box-only` view and get the hierarchy to the root view.